tensorboard/components/plugin_util/message.ts

```diff
@@ -40,21 +46,14 @@ interface PromiseResolver {
   reject: (error: Error) => void;
 }
 
-export abstract class IPC {
-  private idPrefix: string;
+export class IPC {
   private id = 0;
-  private readonly responseWaits = new Map<string, PromiseResolver>();
+  private readonly responseWaits = new Map<number, PromiseResolver>();
   private readonly listeners = new Map<MessageType, MessageCallback>();
 
-  constructor() {
-    window.addEventListener('message', this.onMessage.bind(this));
-
-    // TODO(tensorboard-team): remove this by using MessageChannel.
-    const randomArray = new Uint8Array(16);
-    window.crypto.getRandomValues(randomArray);
-    this.idPrefix = Array.from(randomArray)
-      .map((int: number) => int.toString(16))
-      .join('');
+  constructor(private port: MessagePort) {
+    this.port = port;
+    port.addEventListener('message', this.onMessage.bind(this));
```

Comment:
`bind` is a bit ew unless it is required. consider using fancy arrow function?

Reply:
Done